rest_framework/pagination.py

```diff
@@ -201,7 +201,34 @@ def to_html(self):  # pragma: no cover
         raise NotImplementedError('to_html() must be implemented to display page controls.')
 
 
-class PageNumberPagination(BasePagination):
+class BasePageSizePagination(BasePagination):
```

Comment:
Rather than introduce a new base class I'd prefer us to replicate `get_page_size` directly on `CursorPagination`.
More verbosity but less complexity.